### DISCUSSION TITLE
Safe Exam Browser configurations for notebooks (jupyter, marimo, Pluto)

### DISCUSSION THREAD

**Discussion by VivaldoMendes** [SafeExamBrowser/seb-win-refactoring#1046] [Q&A]
Can I load a Python notebook (e.g., a Jupyter one, a Marimo.py notebook,  or a Julia notebook created by Pluto.jl) under the safe umbrella of the Safe Exam Browser software? By locking Moodle into the Safe Exam Browser software, we can run a test/exam on the Moodle platform and guarantee that there is no cheating in the process. Can we perform a similar task by combining a marimo.py notebook or a Pluto.jl notebook (both of which require the Python or the Julia languages to be already running) and the Safe Exam Browser? Thanks for your help.

**Comment by dbuechel** [comment]
That should indeed be possible, using the possibility to allow third-party applications in SEB as described in the documentation: https://safeexambrowser.org/windows/win_usermanual_en.html#ApplicationsPane.

**Comment by VivaldoMendes** [comment]
Thanks for providing feedback. I have raised issues next to forums associated with Pluto.jl, Marimo.py, and Jupyter.ipynb, and people are unaware of anyone who has implemented that configuration so far. By any chance, do you know of someone who has tried that integration before?

**Comment by danschlet** [comment]
@VivaldoMendes looks like at our university Jupyter Notebooks are used inside of virtual desktops (we're using VMware Horizon VDI), where a Jupyter server is running and students are accessing it via a locked down Chrome browser. SEB secures the virtual desktop client, so students cannot switch to local applications while they are doing the exam in the VDI. Also they seem to be developing a Jupyter Hub integration in Moodle.

But in general we from the SEB team don't know anything about those systems with these funny names you are mentioning. Maybe someone else from the SEB user community tried some integration with Jupyter et all, then hopefully they would reply here. 

Otherwise I would suggest to just try it out yourself. SEB can run local applications secured with its lockdown/kiosk mode, but it cannot secure internet access in those permitted applications/processes and access to the local file system is also possible in those applications. That's the only information we from the SEB developer side can give you.

**Comment by VivaldoMendes** [comment]
Thank you very much for your information. It is indeed handy, and it is good to know that someone did it before. I will have a go and see if I can replicate the procedure implemented at your university.
